docs/source/tooling/truffle.rst

```diff
@@ -1,59 +1,166 @@
-MythX Security Analysis plugin for Truffle
-==========================================
+.. _tooling.truffle:
 
-MythX Security Analysis plugin for Truffle adds automated smart contract security analysis to the `Truffle framework <https://truffleframework.com>`_.
+Truffle
+=======
 
-Prerequisites
-^^^^^^^^^^^^^
+**MythX for Truffle** is a `Truffle plugin <https://truffleframework.com/docs/truffle/getting-started/writing-external-scripts#third-party-plugin-commands>`_ that adds automated smart contract security analysis to the `Truffle framework <https://truffleframework.com>`_. With this plugin, you can run security analysis directly from your Truffle development environment.
 
-The MythX plugin is compatible with Truffle 5.0 or higher. Note that your Truffle project must compile
+The MythX plugin requires Truffle 5.0 or higher. Note that your Truffle project must compile
 successfully for the security analysis to work.
 
-Quickstart
+Installation
+------------
+
+You can install the plugin on a per-project basis or globally.
+
+.. warning:: 
+
+   **Windows users** have reported potential issues with installation. You may have to install the following dependencies:
+
+   * `Git <https://git-scm.com/>`_
+   * `Windows Build Tools npm package <https://www.npmjs.com/package/windows-build-tools>`_::
+
+       npm install --global --production windows-build-tools
+
+     (This will need to be installed with Adminstrator privileges.)
+
+Individual project
+^^^^^^^^^^^^^^^^^^
+
+Install the plugin on an individual Truffle project by running the following inside the root of your Truffle project::
+
+  npm install truffle-security
+
+The plugin will install for that Truffle project only. In addition, the plugin will **edit the project's configuration file** (``truffle-config.js``) to add the necessary plugin configuration. You do not need to edit this file.
+
+.. note:: If you have existing plugins activated for the project, they will not be affected.
+
+Global installation
+^^^^^^^^^^^^^^^^^^^
+
+Install the plugin globally so that it is accessible to all projects::
+
+  npm install truffle-security
+
+If you install MythX for Truffle in this manner, **you will in addition need to edit each project's configuration file** (``truffle-config.js``) to add the necessary plugin:
```

Comment:
We should also mention here that if you run `npm install truffle-security` in the root of your Truffle project, we will automatically detect the config file and add ourselves as a plugin to the `module.exports.plugins` list. That way you can just get started by running two commands (installation and `truffle run verify`) and you're set.

Reply:
@dmuhs I thought I effectively said that here:

> The plugin will install for that Truffle project only. In addition, the plugin will **edit the project's configuration file** (``truffle-config.js``) to add the necessary plugin configuration. You do not need to edit this file.

Did you want me to mention specifically `module.exports.plugins`? Or something else?

Reply:
Whoops, my bad, I misread :smile: